File: SoccerTeamManagement/Data/Models/Player.cs
﻿using SoccerTeamManagement.Data.Models.Joins;
using System.Collections.Generic;

namespace SoccerTeamManagement.Data.Models
{
    public class Player : Person
    {
        #region Properties

        public int? Height { get; set; }

        public int? Weight { get; set; }

        public Enums.Foot Foot { get; set; }

        public Enums.StarRating WeakFootRating { get; set; }

        public Enums.StarRating FlareRating { get; set; }

        #endregion Properties

        #region Relationships

        public int? NationId { get; set; }

        public NationLookup Nation { get; set; }

        public int? AttributesId { get; set; }

        public PlayerAttributes Attributes { get; set; }

        public ICollection<PlayerPosition> PlayerPositions { get; set; }

        public ICollection<TeamPlayer> TeamPlayers { get; set; }

        public ICollection<PlayerParent> PlayerParents { get; set; }

        #endregion Relationships
    }
}

File: SoccerTeamManagement/ClientApp/src/app/interfaces/player.ts
import { IPerson } from "./person";
import { IPlayerAttributes } from "./player-attributes";
import { IParent } from "./Parent";
import { IPlayerPosition } from "./joins/player-position";
import { ITeamPlayer } from "./joins/team-player";

export interface IPlayer extends IPerson {
  height: number;
  weight: number;
  foot: number;
  weakFootRating: number;
  flareRating: number;

  attributesId: number;
  attributes: IPlayerAttributes;
  teamPlayers: ITeamPlayer[];
  playerPositions: IPlayerPosition[];
  parents: IParent[];
}
